plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -73,6 +73,9 @@ type predicateMetadata struct {
 	podRequest                *schedulercache.Resource
 	podPorts                  map[int]bool
 	matchingAntiAffinityTerms []matchingPodAntiAffinityTerm
+	// Service pods cache, otherwise this is a large query, will slow things down
+	// when there are 100s of services/pods.
+	servicePods []*api.Pod
```

Comment:
predicateMeta.servicePods caster out of meta,  are you saying that the modification to the pointer won't persist?